pkg/informer/supportResources_test.go

```diff
@@ -0,0 +1,216 @@
+			},
+		}),
+			namespace: "open-cluster-management",
+			name:      "wrong-name",
+			err:       fmt.Errorf("configmaps \"search-collector-config\" not found")},
+	}
+
+	for _, clientset := range configmaps {
+
+		config, err := clientset.clientset.CoreV1().ConfigMaps(clientset.namespace).Get(context.TODO(), clientset.name, metav1.GetOptions{})
+		if err != nil {
+			if clientset.err == nil { //if config has err but the clientset does not
+				t.Fatalf(err.Error())
+			}
+			if !strings.EqualFold(clientset.err.Error(), err.Error()) { //if the errors don't match
+				t.Fatalf("expected err was %s but got err like %s", clientset.err, err)
+			}
+
+		}
+
+		_, _, allowerr, denyerr := GetAllowDenyData(config)
+
+		if allowerr != nil {
+			if clientset.err == nil {
+				t.Fatalf(allowerr.Error())
+			}
+			if !strings.EqualFold(clientset.err.Error(), allowerr.Error()) {
+				t.Fatalf("expected err was %s but got err like %s", clientset.err, allowerr)
+			}
+		}
+		if denyerr != nil {
+			if clientset.err == nil {
+				t.Fatalf(denyerr.Error())
+			}
+			if !strings.EqualFold(clientset.err.Error(), denyerr.Error()) {
+				t.Fatalf("expected err was %s but got err like %s", clientset.err, denyerr)
+			}
+		}
+
+	}
+
+}
+
+func Test_supportedResources(t *testing.T) {
+
+	clientset := fake.NewSimpleClientset(&v1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "search-collector-config",
+			Namespace: "open-cluster-management",
+		},
+		Data: map[string]string{
+			"AllowedResources": "- apiGroups: \n    - \"*\"\n  resources: \n    - services\n    - pods\n- apiGroups:\n    - authentication.k8s.io\n  resources:\n    - \"*\"",
+			"DeniedResources":  "- apiGroups:\n    - \"*\"\n  resources:\n    - secrets\n- apiGroups:\n    - admission.k8s.io\n  resources:\n    - policies\n    - iampolicies\n    - certificatepolicies\n- apiGroups:\n    - admission.k8s.io\n  resources:\n    - \"*\"",
+		},
+	})
+
+	config, _ := clientset.CoreV1().ConfigMaps("open-cluster-management").Get(context.TODO(), "search-collector-config", metav1.GetOptions{})
+	allow, deny, _, _ := GetAllowDenyData(config)
+
+	//testing allowed:
+	incomingResources := make(map[string]string)
+	incomingGroup := make(map[string]string)
+
+	//testing for any resources:
+	incomingGroup["group"] = "authentication.k8s.io"
+	incomingResources["resource"] = "deployments"
+
+	//testing for any apigroup:
+	incomingGroup["group"] = "anygroup.io"
+	incomingResources["resource"] = "services"
+
+	for _, rec := range incomingResources {
+		for _, group := range incomingGroup {
+			if rec != "" && group != "" {
+
+				if isResourceAllowed(group, rec, allow, deny) != true {
+
+					t.Error("Not expected. Error", group, rec)
+				}
+			} else {
+				t.Error("Not expected. Error")
+			}
+
+		}
+
+	}
```

Comment:
I'd remove the map and just pass the values directly like this.

```go
// Test all resources allowed for group
allowed,_,_ := isResourceAllowed("authentication.k8s.io", "deployments", allow, deny)
if !allowed {
  t.Error("Expected resource deployments.authentication.k8s.io to be allowed, but got denied.")
}

// Test resource matching any group
allowed,_,_ := isResourceAllowed("anygroup.io", "services", allow, deny)
if !allowed {
  t.Error("Expected resource services.anygroup.io to be allowed, but got denied.")
}

// Continue the same pattern with other tests.
}

Reply:
Wait, the above makes it clear that we are testing `isResourceAllowed()` with different inputs, but it creates a fair amount of code duplication.  Let me try to refactor again...

Reply:
that's much simpler!

Reply:
How about this?
```go
testData := [][]string{{"authentication.k8s.io", "deployments"}, {"services.anygroup.io", "services"}}

for data := range testData {
  allowed := isResourceAllowed(data[0], data[1], allow, deny)
  if !allowed {
    t.Error("Expected group: %s resource: %s to be allowed, but got denied.", data[0], data[1])
  }
}
```